File: java/src/test/java/com/alxarafe/domain/topology/valueobject/NamingPolicyTest.java
package com.alxarafe.domain.topology.valueobject;

import com.alxarafe.app.domain.topology.valueobject.NamingPolicy;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class NamingPolicyTest {

    @Test
    void create() {
        var policy = new NamingPolicy("-", 2, 3, 2, 3, 2);
        assertEquals("-", policy.codeSeparator());
        assertEquals(2, policy.warehousePadding());
        assertEquals(3, policy.zonePadding());
        assertEquals(2, policy.aislePadding());
        assertEquals(3, policy.bayPadding());
        assertEquals(2, policy.levelPadding());
    }

    @Test
    void nullSeparator() {
        assertThrows(NullPointerException.class, () -> new NamingPolicy(null, 2, 3, 2, 3, 2));
    }

    @Test
    void blankSeparator() {
        assertThrows(IllegalArgumentException.class, () -> new NamingPolicy("", 2, 3, 2, 3, 2));
    }

    @Test
    void separatorTooLong() {
        assertThrows(IllegalArgumentException.class, () -> new NamingPolicy("----", 2, 3, 2, 3, 2));
    }

    @Test
    void paddingTooLow() {
        assertThrows(IllegalArgumentException.class, () -> new NamingPolicy("-", 0, 3, 2, 3, 2));
    }

    @Test
    void paddingTooHigh() {
        assertThrows(IllegalArgumentException.class, () -> new NamingPolicy("-", 11, 3, 2, 3, 2));
    }
}

File: php/tests/Unit/Domain/Topology/ValueObject/NamingPolicyTest.php
<?php

declare(strict_types=1);

namespace Tests\Unit\Domain\Topology\ValueObject;

use Alxarafe\App\Domain\Topology\ValueObject\NamingPolicy;
use InvalidArgumentException;
use PHPUnit\Framework\TestCase;

final class NamingPolicyTest extends TestCase
{
    public function testCreate(): void
    {
        $policy = new NamingPolicy('-', 2, 3, 2, 3, 2);
        self::assertSame('-', $policy->codeSeparator());
        self::assertSame(2, $policy->warehousePadding());
        self::assertSame(3, $policy->zonePadding());
        self::assertSame(2, $policy->aislePadding());
        self::assertSame(3, $policy->bayPadding());
        self::assertSame(2, $policy->levelPadding());
    }

    public function testEmptySeparator(): void
    {
        $this->expectException(InvalidArgumentException::class);
        new NamingPolicy('', 2, 3, 2, 3, 2);
    }

    public function testSeparatorTooLong(): void
    {
        $this->expectException(InvalidArgumentException::class);
        new NamingPolicy('----', 2, 3, 2, 3, 2);
    }

    public function testPaddingTooLow(): void
    {
        $this->expectException(InvalidArgumentException::class);
        new NamingPolicy('-', 0, 3, 2, 3, 2);
    }

    public function testPaddingTooHigh(): void
    {
        $this->expectException(InvalidArgumentException::class);
        new NamingPolicy('-', 11, 3, 2, 3, 2);
    }

    public function testEquals(): void
    {
        $a = new NamingPolicy('-', 2, 3, 2, 3, 2);
        $b = new NamingPolicy('-', 2, 3, 2, 3, 2);
        $c = new NamingPolicy('.', 2, 3, 2, 3, 2);
        self::assertTrue($a->equals($b));
        self::assertFalse($a->equals($c));
    }
}
